Adding the "any" pseudo-device since Solaris supports it.
This is an in-house patch and will eventually be sent upstream.

--- a/pcap-bpf.c	2016-10-25 17:07:59.000000000 -0700
+++ b/pcap-bpf.c	2017-10-03 16:14:50.480166759 -0700
@@ -2661,7 +2661,17 @@
 	 */
 	if (pcap_findalldevs_interfaces(alldevsp, errbuf, check_bpf_bindable) == -1)
 		return (-1);	/* failure */
-
+ 
+#ifdef DLT_IPNET 
+	/*
+	 * Adding the "any" pseudo-device
+	 */
+	const char any_descr[] = "Pseudo-device that captures on all interfaces";
+	if (pcap_add_if(alldevsp, "any", IFF_UP|IFF_RUNNING,
+	    any_descr, errbuf) < 0)
+		return (-1);
+#endif
+ 
 #if defined(__FreeBSD__) && defined(SIOCIFCREATE2)
 	if (finddevs_usb(alldevsp, errbuf) == -1)
 		return (-1);
